src/main/java/net/dries007/tfc/util/PowderKegExplosion.java

```diff
@@ -77,7 +80,11 @@ public void finalizeExplosion(boolean spawnParticles)
             this.level.addParticle(ParticleTypes.EXPLOSION_EMITTER, this.x, this.y, this.z, 1.0D, 0.0D, 0.0D);
         }
 
-        final List<BlockPos> affectedBlockPositions = this.getToBlow();
+        List<BlockPos> affectedBlockPositions = this.getToBlow()
```

Comment:
Sorry to make you change things again -- should have been clearer.

This is doubling the amount of getBlockState calls per explosion as well as using a stream which has minor performance implications.

It should be like so

```java
for (BlockPos pos : affectedBlockPositions)
  {
    final BlockState state = level.getBlockState(pos);
    
    if (Helpers.isBlock(state, TFCTags.Blocks.EXPLOSION_PROOF))
        continue;
     
  }```